src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,224 @@
+package mate.jdbc.dao;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import mate.jdbc.exception.DataProcessingException;
+import mate.jdbc.lib.Dao;
+import mate.jdbc.model.Car;
+import mate.jdbc.model.Driver;
+import mate.jdbc.model.Manufacturer;
+import mate.jdbc.util.ConnectionUtil;
+
+@Dao
+public class CarDaoImpl implements CarDao {
+
+    @Override
+    public Car create(Car car) {
+        String query = "INSERT INTO `library_db`.`cars` (model, manufacture_id) VALUES (?, ?);";
+        Long manufacturerId = getManufacturerId(car.getManufacturer().getName());
+
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement createCarStatement
+                        = connection.prepareStatement(query, Statement.RETURN_GENERATED_KEYS)) {
+            createCarStatement.setString(1, car.getModel());
+            createCarStatement.setObject(2, manufacturerId);
+            createCarStatement.executeUpdate();
+            ResultSet resultSet = createCarStatement.getGeneratedKeys();
+            if (resultSet.next()) {
+                car.setId(resultSet.getObject(1, Long.class));
+            }
+            return car;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't create " + car + ". ", e);
+        }
+    }
+
+    private Long getManufacturerId(String name) {
+        String query = "SELECT `id` FROM `library_db`.`manufacturers` WHERE name = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getManufacturerIdStatement
+                        = connection.prepareStatement(query)) {
+            getManufacturerIdStatement.setString(1, name);
+            ResultSet resultSet = getManufacturerIdStatement.executeQuery();
+            Long id = null;
+            if (resultSet.next()) {
+                id = resultSet.getObject(1, Long.class);
+            }
+            return id;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't get id for " + name
+                    + " from manufacturers");
+        }
+    }
+
+    @Override
+    public Car get(Long id) {
```

Comment:
let's return optional from this method so we will give more flexibility to client that will use this code